objects/Wall.js

```diff
@@ -0,0 +1,71 @@
+import {defs, tiny} from '../examples/common.js';
+import CustomObject from './CustomObject.js';
+
+const {
+    Vector, Vector3, vec, vec3, vec4, color, hex_color, Shader, Matrix, Mat4, Light, Shape, Material, Scene,
+} = tiny;
+
+const {Cube} = defs;
+
+// Wall: class that renders wall & window
+export default class Wall extends CustomObject {
+    constructor() {
+        super();
+        this.shapes = {
+            wall_bottom: new Cube(),
+            wall_middle: new Cube(),
+            wall_top: new Cube,
+            window: new Cube(),
+            train_map: new Cube(),
+            disp: new Cube(),
+            poster: new Cube(),
+        };
+
+        this.materials = {
+            test: new Material(new defs.Phong_Shader(),
+                {ambient: .4, diffusivity: .6, color: hex_color("#ffffff")}),
+        }
+    }
+
+    /* Custom object functions */
+    render(context, program_state, window_height=2.35, window_width=5, model_transform=Mat4.identity()) {
+        const wall_bottom_height=1.75;
+        const wall_middle_height=2.3;
+        const wall_top_height=1.5
+        const wall_width=20;
+
+        const map_width=2;
+        const map_height=1;
+        const disp_width=2.8;
+        const disp_height=1;
+        const poster_width=1.5;
+        const poster_height=1;
+
+        const wall_bottom_transform = model_transform.times(Mat4.scale(wall_width, wall_bottom_height, 0.1)).times(Mat4.translation(0,0.95,-15));
+        const wall_middle_transform = model_transform.times(Mat4.scale(wall_width, wall_middle_height, 0.1)).times(Mat4.translation(0, 2.48, -15));
+        const wall_top_transform = model_transform.times(Mat4.scale(wall_width, wall_top_height, 0.1)).times(Mat4.translation(0, 6.3,-15));
+        const window_transform = model_transform.times(Mat4.scale(window_width, window_height, 0.1)).times(Mat4.translation(0,2.3,-14));
```

Comment:
Unless it was intentional, I believe these transforms are backwards. Applying scaling before translation changes
the units that the translation happens in. That is why the translation matrices have very odd numbers to translate the wall by on each axis. We should switch the order to get a translation in standard cartesian coordinates before applying a scale that changes the coordinate base. It'll make it easier to adjust the dimensions later. 

I also notice that the wall is translated backwards by -1.5 units (-15 of 0.1 units). It would probably be easiest to keep it centered around the z-axis at 0. Then later we can translate it backwards or forwards as needed through the `model_transform` parameter when we call it in `main.js`.